### PR TITLE
fix(Core/Raid): Ulduar - Kologarn

### LINKED ISSUE #1781: Kologarn in Ulduar Casts Petrifying Breath too early
Kologarn in Ulduar Casts Petrifying Breath too early

##### SMALL DESCRIPTION:

Kologarn in Ulduar Casts Petrifying Breath as soon as he is pulled, this needs a delay else it causes a raid wipe.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Kologarn in Ulduar has a small delay before casting Petrifying Breath as soon as he is pulled

##### CURRENT BEHAVIOUR:

Kologarn in Ulduar Casts Petrifying Breath as soon as he is pulled, this needs a delay else it causes a raid wipe.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Attack kologarn in ulduar

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

15ffcc65aa1e5f59df3839203f190e55f78b8bd2

##### OPERATING SYSTEM:

ubuntu 16.04

##### MODULES:

No

##### OTHER CUSTOMIZATIONS:

No

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/73470611-kologarn-in-ulduar-casts-petrifying-breath-too-early?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Pondaveia**
- Doors not closing or opening during and after encounter
- Petrifying gaze should have a start window so the raid doesn't wipe on pull

##### CHANGES PROPOSED:

-  Doors now close during encounter and open when over/reset
-  Petrifying gaze now has a starting window so the raid doesn't wipe when pulling

###### ISSUES ADDRESSED:

Closes #1781 

##### HOW TO TEST THE CHANGES:

.tele ulduar
. kill XT and Flame Leviathan
. go to kologarn chamber
.engage

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Dreaxxx** [comment]
Not work properly for me , Doors close fine , , breath not appear in  pull , but the boss doesn't use anymore all others abilities

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_kologarn.cpp [hunk 1]
```diff
@@ -372,6 +385,9 @@ class boss_kologarn : public CreatureScript
 
             switch (events.GetEvent())
```
Comment (Pondaveia) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_kologarn.cpp]:
```suggestion
            switch (events.ExecuteEvent())
```

**Comment by Pondaveia** [comment]
@Dreaxxx can you test again if you will?

**Comment by Dreaxxx** [comment]
Now , the boss use only 1 time each abilities and after that , only auto attacks

**Comment by Naxxramas-Server** [comment]
Ok i maybe the problem is the ScheduleEvent , we need to make ReScheduleEvent no ?

**Comment by Pondaveia** [comment]
@Dreaxxx Thank you for the testing, just realized the overrides stopped the repeats... I'll take care of it asap

**Comment by Pondaveia** [comment]
If it ain't right this time it would probably need some small tunning

**Pull request merged**